pysqream_blue/cursor.py

```diff
@@ -157,66 +165,77 @@ def _compile(self, query):
         self.stmt_id, self.columns_metadata, self.query_type = response.context_id, response.columns, response.query_type
         return response
 
-    def _request_execute(self):
-        try:
-            execute_response: qh_messages.ExecuteResponse = self._execute()
-
-            if execute_response.HasField('error'):
-                self.logs.log_and_raise(OperationalError,
-                              f'Query id: {self.stmt_id}. Error while attempting to execute the query.\n{execute_response.error}')
-
-            self.logs.message(f'Query id: {self.stmt_id}. The query was send to execute successfully.', self.logs.info)
+    def _request_execute_inner(self):
+        execute_response: qh_messages.ExecuteResponse = self._execute()
+        if execute_response.HasField('error'):
+            self.logs.log_and_raise(OperationalError,
+                          f'Query id: {self.stmt_id}. Error while attempting to execute the query.\n{execute_response.error}')
+        self.logs.message(f'Query id: {self.stmt_id}. The query was send to execute successfully.', self.logs.info)
 
-        except grpc.RpcError as rpc_error:
-            self.logs.log_and_raise(ProgrammingError,
-                          f'Query id: {self.stmt_id}. Error from grpc while attempting to execute the query.\n{rpc_error}')
-            self.channel.close()
+    def _request_execute(self):
+        self._retry_function_on_stream_error(self._request_execute_inner, "execute the query")
 
     def _execute(self):
         return self.client.Execute(
             qh_messages.ExecuteRequest(context_id=self.stmt_id),
             credentials=self.call_credentialds if self.use_ssl else None)
 
-    def _request_status(self):
+    def _retry_function_on_stream_error(self, func, function_action_name, *args):
```

Comment:
if we know that the args is a query its better to add query = None and than you can do something like this - 
```
def _retry_function_on_stream_error(self, func, function_action_name, query=None):
    if query is not None:
        return func(query)
    return func()
```
and call like this - 
```
self._retry_function_on_stream_error(func, "some_action", query=query)
self._retry_function_on_stream_error(func, "some_action")

```